esmf_regrid/tests/unit/experimental/unstructured_scheme/test_regrid_unstructured_to_unstructured.py

```diff
@@ -0,0 +1,220 @@
+    result.add_aux_coord(scalar_height)
+    result.add_aux_coord(scalar_time)
+    return result
+
+
+def test_flat_cubes():
+    """
+    Basic test for :func:`esmf_regrid.experimental.unstructured_scheme.regrid_unstructured_to_unstructured`.
+
+    Tests with flat cubes as input (a 1D mesh cube and a 2D grid cube).
+    """
+    src = _flat_mesh_cube()
+
+    n_lons = 6
+    n_lats = 5
+    lon_bounds = (-180, 180)
+    lat_bounds = (-90, 90)
+    tgt = _flat_mesh_cube(n_lons, n_lats, lon_bounds, lat_bounds, circular=True)
+    # Ensure data in the target grid is different to the expected data.
+    # i.e. target grid data is all zero, expected data is all one
+    tgt.data[:] = 0
+
+    src = _add_metadata(src)
+    src.data[:] = 1  # Ensure all data in the source is one.
+    result = regrid_unstructured_to_unstructured(src, tgt)
+
+    expected_data = np.ones([n_lats, n_lons])
+    expected_cube = _add_metadata(tgt)
+
+    # Lenient check for data.
+    assert np.allclose(expected_data, result.data)
+
+    # Check metadata and scalar coords.
+    expected_cube.data = result.data
+    assert expected_cube == result
+
+
+@pytest.mark.parametrize("method", ("bilinear", "nearest"))
+def test_node_friendly_methods(method):
+    """
+    Basic test for :func:`esmf_regrid.experimental.unstructured_scheme.regrid_unstructured_to_unstructured`.
+
+    Tests with the bilinear and nearest method.
+    """
+    n_lons = 6
+    n_lats = 5
+    lon_bounds = (-180, 180)
+    lat_bounds = (-90, 90)
+    src = _gridlike_mesh_cube(n_lons, n_lats, location="node")
+    tgt = _gridlike_mesh_cube(n_lons, n_lats, location="node")
+    # Ensure data in the target mesh is different to the expected data.
+    # i.e. target mesh data is all zero, expected data is all one
+    tgt.data[:] = 0
+
+    src = _add_metadata(src)
+    src.data[:] = 1  # Ensure all data in the source is one.
+    result = regrid_unstructured_to_unstructured(src, tgt, method=method)
+
+    expected_data = np.ones(tgt.data)
+    expected_cube = _add_metadata(tgt)
+
+    # Lenient check for data.
+    assert np.allclose(expected_data, result.data)
+
+    # Check metadata and scalar coords.
+    expected_cube.data = result.data
+    assert expected_cube == result
+
+
+def test_invalid_args():
+    """
+    Test for :func:`esmf_regrid.experimental.unstructured_scheme.regrid_unstructured_to_unstructured`.
+
+    Tests that an appropriate error is raised when arguments are invalid.
+    """
+    n_lons = 6
+    n_lats = 5
+    lon_bounds = (-180, 180)
+    lat_bounds = (-90, 90)
+    node_src = _gridlike_mesh_cube(n_lons, n_lats, location="node")
+    edge_src = _gridlike_mesh_cube(n_lons, n_lats, location="edge")
+    face_src = _gridlike_mesh_cube(n_lons, n_lats, location="face")
+    tgt = _gridlike_mesh_cube(n_lons, n_lats)
+
+    with pytest.raises(ValueError):
+        _ = regrid_unstructured_to_unstructured(tgt, tgt, method="bilinear")
```

Comment:
In the other tests, this would check that the method only works with the grid/mesh type given. Since we have now implemented every type of mesh/grid regridding, I don't think it's necessary to check that such an error is being thrown anymore. I think this line can be removed.